### PR TITLE
Add pyproject.toml

### PR DESCRIPTION
**Pull request by hexagonrecursion**
This blocks #6264 because black configuration is stored in pyproject.toml.
See <https://black.readthedocs.io/en/stable/usage_and_configuration/the_basics.html#configuration-format>
I copied this from <https://setuptools.pypa.io/en/latest/userguide/quickstart.html#basic-use>

### CODE REVIEW

**Comment by hexagonrecursion** [comment]
> Permission denied: '/usr/local/lib/python3.9

I am working on this.

**Comment by hexagonrecursion** [comment]
https://github.com/pypa/setuptools/issues/3019

File: setup.py [hunk 1]
```diff
@@ -19,6 +19,7 @@
 except ImportError:
     cythonize = None
 
+sys.path += [os.path.dirname(__file__)]
```
Comment (ThomasWaldmann) [inline setup.py:22]: so I assume it didn't work without that even if you invoked stuff from the toplevel dir?

usually one should rather try not having to modify sys.path.

**Comment by ThomasWaldmann** [comment]
btw, for now please no bigger / fundamental changes to setup.py / setup.cfg.

File: setup.py (hunk 1 above)
Comment (hexagonrecursion) [reply setup.py:22]: Neither the CWD nor the parent of setup.py are on sys.path if we use setuptools.build_meta. This is, apparently, a feature. Either we append `os.path.dirname(__file__)` or we use `setuptools.build_meta:__legacy__`, which will do the same for us.

https://github.com/pypa/setuptools/discussions/3134#discussioncomment-2249652

> The default backend (setuptools.build_meta) will not add the project folder to the path on purpose.
> That is only supported in `setuptools.build_meta:__legacy__`

Comment (ThomasWaldmann) [reply setup.py:22]: ok, so keep it with line 22. hopefully this won't shoot ourselves in the knee at some time.

**Comment by hexagonrecursion** [comment]
Another workaround would be to `pip install .` instead of `pip install -e .`

**Review by ThomasWaldmann — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**